library/src/main/java/com/mux/video/upload/internal/UploadInfo.kt

```diff
@@ -17,8 +17,11 @@ import java.io.File
  * Job and Flows populated
  */
 internal data class UploadInfo(
```

Comment:
I think this is the result of the crashes. `UploadInfo` in the android SDK is supposed to be immutable, so you'd return a new `UploadInfo` whenever you changed something in your function. The idea was to avoid concurrency issues and keep SDK callers from messing with it (this is visible from java).

This was probably all overkill, so I volunteer to fix the crash. Hopefully by making `UploadInfo` more reasonable to use.

Reply:
Out of curiosity do we have a stack trace or log from one of the crashes?

Reply:
I'm not sure that it will be helpful. The reason this crashes is because the `UploadInfo` object is supposed to emulate an immutable struct, where you'd "mutate" it by returning a new instance. It's (almost) like `struct` vs `class` in swift. 

The way that `TranscodeContext` is supposed to update `UploadInfo` is to use `UploadInfo.update()` to create a new `UploadInfo` and return that from `start()`. This pattern is great for thread safety (the `UploadInfo`'s fields are always consistent), but not intuitive to use in kotlin/java. I'm in the middle of fixing it now, and will probably open a PR against this one